force-app/main/default/lwc/ldsCreateRecord/__tests__/ldsCreateRecord.test.js

```diff
@@ -0,0 +1,138 @@
+        }
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    // Helper function to wait until the microtask queue is empty. This is needed for promise timing when calling UI API
+    function flushPromises() {
+        // eslint-disable-next-line no-undef
+        return new Promise(resolve => setImmediate(resolve));
+    }
+
+    it('sets value from lightning-input field as parameter to createRecord call', () => {
+        const USER_INPUT = 'Gomez Inc.';
+        const INPUT_PARAMETERS = [
+            { apiName: 'Account', fields: { Name: USER_INPUT } }
+        ];
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return Promise.resolve().then(() => {
+            expect(createRecord).toHaveBeenCalled();
+            expect(createRecord.mock.calls[0]).toEqual(INPUT_PARAMETERS);
+        });
+    });
+
+    it('displays an account id after record creation', () => {
+        const USER_INPUT = 'Gomez Inc.';
+        const RESULT = '0011700000r6zdbAAA';
+        createRecord.mockResolvedValue(mockCreateRecord);
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        const inputEl = element.shadowRoot.querySelector(
+            'lightning-input[class="slds-m-bottom_x-small"]'
+        );
+        inputEl.value = USER_INPUT;
+        inputEl.dispatchEvent(new CustomEvent('change'));
+
+        const buttonEl = element.shadowRoot.querySelector('lightning-button');
+        buttonEl.click();
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return flushPromises().then(() => {
+            const displayEl = element.shadowRoot.querySelector(
+                'lightning-input:not([class="slds-m-bottom_x-small"])'
+            );
+            expect(displayEl.value).toBe(RESULT);
+        });
+    });
+
+    it('displays a success toast after record creation', () => {
+        const USER_INPUT = 'Gomez Inc.';
+        createRecord.mockResolvedValue(mockCreateRecord);
+
+        const element = createElement('c-lds-create-record', {
+            is: LdsCreateRecord
+        });
+        document.body.appendChild(element);
+
+        // Mock handler for toast event
+        const handler = jest.fn();
+        // Add event listener to catch toast event
+        element.addEventListener('lightning__showtoast', handler);
```

Comment:
Can't I use `ShowToastEventName` from the `ShowToastEvent` mock? @trevor-bliss

Reply:
Right now it's not exported from the mock. We could export it though...

Reply:
Using an export from my custom mock now.